src/ui/dialogs/multiUserJoin.ts

```diff
@@ -138,7 +143,27 @@ class MultiUserJoinDialog {
          });
 
          return Promise.all(promises).then((results) => {
-            return results.filter(jid => typeof jid !== 'undefined');
+            results=results.filter(jid => typeof jid !== 'undefined');
+            results.sort(  function compare( a, b ) {
+              if (!a.domain||!b.domain)
+                  return 0;
+              if ( a.domain.startsWith('conference') && !b.domain.startsWith('conference')){
+                return -1;
+              }
+              if ( !a.domain.startsWith('conference') && b.domain.startsWith('conference')){
+                return +1;
+              }
+              return a.domain.localeCompare(b.domain);
+            });
+            this.serverInputList.empty();
+            $('#jsxc-serverlist select').empty();
+
+            let i=0;
+            for (;i<results.length;i++) {
+                this.serverInputList.append('<option>'+results[i].domain+'</option>');
+                $('#jsxc-serverlist select').append('<option>'+results[i].domain+'</option>');
```

Comment:
This allows the server to inject code to the application. `$('#jsxc-serverlist select').append($('<option>').text(results[i].domain));` would be saver.

Reply:
changed to your suggestion